pvlib/test/test_solarposition.py

```diff
@@ -499,3 +503,35 @@ def test_analytical_azimuth():
     azimuths = solarposition.solar_azimuth_analytical(*test_angles.T, zenith=zeniths)
 
     assert not np.isnan(azimuths).any()
+
+
+def test_analytical_sunrise_sunset_transit():
+    """Test analytical calculations for sunrise, sunset, and transit times"""
+    times = pd.DatetimeIndex(start='2018-01-01 0:00:00',
+                             end='2018-12-31 23:59:59',
+                             freq='H').tz_localize('Etc/GMT+7')
+    lat, lon = 39.743, -105.178  # degrees
+    eot = solarposition.equation_of_time_pvcdrom(times.dayofyear)  # minutes
+    decl = solarposition.declination_spencer71(times.dayofyear)  # radians
+    sst = solarposition.sunrise_sunset_transit_analytical(
+        times, latitude=lat, longitude=lon, declination=decl,
+        equation_of_time=eot)
+    sunrise = (sr.time() for sr in sst[0])
+    sunset = (ss.time() for ss in sst[1])
+    transit = (tr.time() for tr in sst[2])
+    sunrise_hours = [sr.hour + (sr.minute + sr.second/60.)/60.
+                     for sr in sunrise]
+    sunset_hours = [ss.hour + (ss.minute + ss.second/60.)/60. for ss in sunset]
+    transit_hours = [tr.hour + (tr.minute + tr.second/60.)/60.
+                     for tr in transit]
+    test_data_file = os.path.join(
+        DATADIR, 'sunrise_sunset_transit.txt')
```

Comment:
do we need such a large data file to comprehensively test the algorithm? I am guessing 10-20 well chosen points is sufficient.

Reply:
I guess not, but I thought others might want to use this, it's a full year of hourly sunrise/sunset/transit times from NREL SPA. It's about 600kb, is that too much? Whatever you think is best, maybe ditch it completely, and just test 5-10 values hardcoded in the test?